### DISCUSSION TITLE
Max word limit for punctuation model

### DISCUSSION THREAD

**Discussion by kamrapooja** [NVIDIA-NeMo/NeMo#4612] [General]
What is the maximum sentence length is allowed for punctuation model?

**Comment by kamrapooja** [comment]
Please help

**Comment by evilc3** [comment]
max length for any transformer model will be the max_tokens for Bert its 512, check the lm ur using. But the punctuation model makes use of windowing. So there should be a max length constraint

**Comment by ekmb** [comment]
[This script](https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/token_classification/punctuate_capitalize_infer.py#L47) splits the input into overlapping segments to run inference on long texts and overcome LM max_seq_length constraint.

**Comment by kamrapooja** [comment]
is there any number for long text? We have paragraph of 10000 words. Will it be able to process it?
